cadl-extension/src/emitter.ts

```diff
@@ -2,27 +2,46 @@ import { resolvePath, getNormalizedAbsolutePath, Program } from "@cadl-lang/comp
 import { dump } from "js-yaml";
 import { promisify } from "util";
 import { execFile } from "child_process";
-import { mkdir } from "fs";
+import { promises } from "fs";
 import { CodeModelBuilder } from "./code-model-builder.js";
 
 export async function $onEmit(program: Program) {
   const builder = new CodeModelBuilder(program);
   const codeModel = builder.build();
 
+  const tempConfigurationFile = resolvePath("./java.json");
+  let tempConfigurationFileAccessible = true;
+  await promises.access(tempConfigurationFile).catch((_) => {
+    tempConfigurationFileAccessible = false;
+  });
+  if (tempConfigurationFileAccessible) {
+    const configurationContent = await program.host.readFile(tempConfigurationFile);
+    const configuration = JSON.parse(configurationContent.text);
+
+    program.logger.debug(`Configuration file (temporary) content: ${configurationContent.text}`);
+
+    (codeModel as any).configuration = configuration;
+  }
+
   const outputPath = program.compilerOptions.outputPath ?? getNormalizedAbsolutePath("./cadl-output", undefined);
   const codeModelFileName = resolvePath(outputPath, "./code-model.yaml");
 
-  await promisify(mkdir)(outputPath).catch((err) => {
+  await promises.mkdir(outputPath).catch((err) => {
     if (err.code !== "EISDIR" && err.code !== "EEXIST") {
       throw err;
     }
   });
 
   await program.host.writeFile(codeModelFileName, dump(codeModel));
 
+  program.logger.info(`Code model file written to ${codeModelFileName}`);
+
+  const jarFile = "node_modules/@azure-tools/cadl-java/target/azure-cadl-extension-jar-with-dependencies.jar";
+  program.logger.info(`Exec JAR ${jarFile}`);
+
   const output = await promisify(execFile)("java", [
```

Comment:
why some places use `promises`, some places use `promisify`?

Reply:
`fs.promises` is the native promise support in `fs`, the built-in lib.

`promisify` is not built-in lib, but AFAIK, `child_process` does not have something equvalent to `fs.promises`. Let me know if this is not the case.